examples/context/src/custom-inject-resolve.ts

```diff
@@ -24,7 +25,7 @@ const resolve: ResolverFunction = (
   session: ResolutionSession,
 ) => {
   console.log('Context: %s Binding: %s', ctx.name, session.currentBinding!.key);
-  const targetName = ResolutionSession.describeInjection(injection).targetName;
+  const targetName = describeInjection(injection).targetName;
```

Comment:
This is a breaking change from API perspective. Is it really worth the effort? If so, let's keep `ResolutionSession.describeInjection()`.

Reply:
As a quick reply, We may keep it using (there is a way to do it in a typescript way)
```
ResolutionSession.describeInjection = describeInjection
```
if backward compatibility is the main concern. There is a huge cognitive load this function being the static function brings.
I will update this message with more thorough arguments very soon. 
In short: 
1) It is unrelated to the class it is defined on (it doesn't contain a single instance of that class)
2) Its implementation also doesn't use other static functions
3) It breaks symmetry inside ResolutionSession class, which I am trying to expose to use it as a base for refactoring
4) It doesn't seem to be used much outside core (but I might be wrong here -- please correct me if I am wrong)

The refactoring mentioned in (3) seems to be completely equivalent to the current implementation 
in cases where it applies, but allows for parallel resolution, avoiding forks, a more clear mental model, and fewer moving parts that can break locally (ie: push or pop additional time and the entire stack is messed up) and ruin the whole resolution.

Reply:
Another way to preserve compatibility is just have a free function and single line static method that simply calls it. Possibly deprecate it in favour of free function when semantic versioning allows it. Sounds good?